source/extensions/filters/http/basic_auth/basic_auth_filter.cc

```diff
@@ -111,6 +111,7 @@ Http::FilterHeadersStatus BasicAuthFilter::onDenied(absl::string_view body,
       Http::Code::Unauthorized, body,
       [this](Http::ResponseHeaderMap& headers) {
         const auto request_headers = this->decoder_callbacks_->requestHeaders();
+        RELEASE_ASSERT(request_headers.has_value(), "request_headers has no value.");
```

Comment:
Please add a comment why this should always be true, and consider changing to ASSERT.
I'm assuming that `onDenied()` cannot be invoked before receiving the headers (e.g., timeout or something else), but this should be clarified here.